core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProviderSpec.scala

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.dsde.rawls.entities.datarepo
+
+import java.util.UUID
+
+import bio.terra.workspace.model.DataReferenceDescription.ReferenceTypeEnum
+import org.broadinstitute.dsde.rawls.entities.exceptions.DataEntityException
+import org.broadinstitute.dsde.rawls.model.EntityTypeMetadata
+import org.scalatest.{AsyncFlatSpec, Matchers}
+import spray.json.{JsArray, JsNumber, JsObject, JsString}
+
+class DataRepoEntityProviderSpec extends AsyncFlatSpec with DataRepoEntityProviderSpecSupport with Matchers {
+
+  behavior of "DataEntityProvider.entityTypeMetadata()"
+
+  it should "return entity type metadata in the golden path" in {
+    val provider = createTestProvider()
+
+    provider.entityTypeMetadata() map { metadata: Map[String, EntityTypeMetadata] =>
+      // this is the default expected value, should it move to the support trait?
+      val expected = Map(
+        ("table1", EntityTypeMetadata(-1, "", Seq("col1.1", "col1.2"))),
+        ("table2", EntityTypeMetadata(-1, "", Seq("col2.1", "col2.2"))))
+      assertResult(expected) { metadata }
+    }
+  }
+
+  it should "return an empty Map if data repo snapshot has no tables" in {
+    val provider = createTestProvider(
+      dataRepoDAO = new SpecDataRepoDAO(Right( createSnapshotModel( List.empty[(String, List[String])] ) )))
+
+    provider.entityTypeMetadata() map { metadata: Map[String, EntityTypeMetadata] =>
+      assert(metadata.isEmpty, "expected response data to be the empty map")
+    }
+  }
+
+  it should "return empty columns if data repo snapshot tables but no columns" in {
+    val provider = createTestProvider(
+      dataRepoDAO = new SpecDataRepoDAO(Right( createSnapshotModel( List( ("foo", List.empty[String]), ("bar", List("one", "two")) ) ) )))
+
+    val expected = Map(
+      ("foo", EntityTypeMetadata(-1, "", Seq.empty[String])),
+      ("bar", EntityTypeMetadata(-1, "", Seq("one", "two"))))
+
+    provider.entityTypeMetadata() map { metadata: Map[String, EntityTypeMetadata] =>
+      assertResult(expected) { metadata }
+    }
+  }
+
+  it should "bubble up error if workspace manager errors" in {
+    val provider = createTestProvider(
+      workspaceManagerDAO = new SpecWorkspaceManagerDAO(Left(new bio.terra.workspace.client.ApiException("whoops 1"))))
+
+    val ex = intercept[bio.terra.workspace.client.ApiException] { provider.entityTypeMetadata() }
+    assertResult("whoops 1") { ex.getMessage }
+  }
+
+  it should "bubble up error if data repo errors" in {
+    val provider = createTestProvider(
+      dataRepoDAO = new SpecDataRepoDAO(Left(new bio.terra.datarepo.client.ApiException("whoops 2"))))
+
+    val ex = intercept[bio.terra.datarepo.client.ApiException] { provider.entityTypeMetadata() }
+    assertResult("whoops 2") { ex.getMessage }
+  }
+
+  // to-do: tests for different primary key values returned by data repo, once TDR supports this
```

Comment:
How do we know when TDR starts supporting these? We should consider creating tickets for these tests (if they don't already exist) and linking them to the corresponding ticket on TDR team's JIRA.

Reply:
updated comments.

Reply:
Looks like DR-1003 was completed yesterday? I'm not 100% sure what 'Done' means for their team, so adding "tests for entity/row counts returned by data repo" could be a part of a future PR.

Reply:
I saw that the jira ticket was marked Done, I did not see actual row count data in the Data Repo dev instance, so I haven't adopted it yet, I'll make a follow-on AS ticket to track this work.

Reply:
tracking as AS-365